### PR TITLE
[jit] add index and count to list

### PR DESCRIPTION
**Pull request by TheCodez**
see https://github.com/pytorch/pytorch/issues/16662

### CODE REVIEW

File: torch/csrc/jit/register_prim_ops.cpp [hunk 1]
```diff
@@ -1396,6 +1464,10 @@ RegisterOperators reg2({
 
     Operator("aten::remove(Tensor[](a!) self, Tensor el) -> ()",
         listRemove<Shared<TensorList>, at::Tensor>),
+    Operator("aten::index(Tensor[] self, Tensor el) -> int",
```
Comment (fmassa) [inline torch/csrc/jit/register_prim_ops.cpp]: We already have an `aten::index` which performs advanced indexing on tensors. Would it be a problem having conflicting names here?

Comment (TheCodez) [reply torch/csrc/jit/register_prim_ops.cpp]: The jit tests ran successfully on my machine. However, it doesn't seem like a good idea to have two completely different operations share the same name.

Comment (suo) [reply torch/csrc/jit/register_prim_ops.cpp]: It should be fine, we have overload resolution logic that will select the correct version.

File: test/test_jit.py [hunk 2]
```diff
@@ -4179,6 +4179,76 @@ def test_list_remove():
...
+        def test_list_index_not_existing():
+            a = [4, 1, 3, 2]
+            i = a.index(5)
+
+            return i
+
+        with self.assertRaisesRegex(RuntimeError, "'5' is not in list"):
+            test_list_index_not_existing()
+
+    def test_mutable_list_index(self):
```
Comment (Krovatkin) [inline test/test_jit.py]: a very **minor** nitpick: i wonder if `mutable` qualifier adds any value here? Aren't all `list`s mutable in python, we could just use `int_list`, `tensor_list` unless you are referring to an underlying element type, then `int`s should be immutable?

**Review by Krovatkin — APPROVED** [review]
looks gut to me.

File: test/test_jit.py (hunk 2 above)
Comment (TheCodez) [reply test/test_jit.py]: @Krovatkin I changed the name of the tests.

Comment (Krovatkin) [reply test/test_jit.py]: thank you 😄 !

Comment (eellison) [reply test/test_jit.py]: Not that it needs changing again, but on the other hand the existing convention  these new tests were following the convention already set in test_jit.py, like with test_mutable_list_append. It also makes easier to grep for mutable changes. (again not that it needs to be rechanged).

**Comment by Krovatkin** [comment]
@TheCodez Could you please merge https://github.com/TheCodez/pytorch/pull/1/ ? This hopefully takes care of macosx comp errors.

**Review by Krovatkin — CHANGES_REQUESTED** [review]
let's fix all the build errors first.

**Comment by Krovatkin** [comment]
@TheCodez Thanks!

**Review by Krovatkin — APPROVED** [review]
@TheCodez , I'll try to merge this today.

**Pull request closed without merging**